ACMILabs/works-embeddings
# #12 Add image embeddings

Resolves #12

Let's add image embeddings so that we can get similar works based on their images/videos.

### Acceptance Criteria
- [x] Add image embeddings using [OpenCLIP](https://github.com/mlfoundations/open_clip)

### Relevant design files
* None

### Testing instructions
1. Update your `config.env` to set `REBUILD=true`, `PAGES=10` and XOS to the dev server: `XOS_API_ENDPOINT=http://dev.acmi.net.au:1005/api/` with `AUTH_TOKEN=<your-dev-server-token>`
1. Build the environment: `make build` (it will take a while to download the OpenCLIP model and rebuild the vector database)
1. Search for work images of people wearing hats: http://localhost:8081/images/?text=people+wearing+hats
1. See similar images works: http://localhost:8081/images/?work=image_1110_0.0
1. Search for video frames of people wearing hats: http://localhost:8081/videos/?text=people+wearing+hats
1. See similar video frames works: http://localhost:8081/videos/?work=video_2689_924.0
